validation.js

```diff
@@ -19,3 +21,23 @@ form.addEventListener('submit', (event) => {
   }
   event.preventDefault();
 });
+
+function saveInfo() {
+  const userInfo = JSON.stringify({
+    username: userName.value,
+    email: email.value,
+    text:
+     text.value,
+  });
+  localStorage.setItem('userInfo', userInfo);
+}
+
+userName.addEventListener('input', saveInfo);
+email.addEventListener('input', saveInfo);
+text.addEventListener('input', saveInfo);
```

Comment:
[OPTIONAL] To improve the code and follow the `DRY` principles, consider creating a helper function here 🤔 it could help if you need to have other fields and you don't want to keep repeating the same things. It could be something like:
```
const saveAllInfos = (elements, function) => {
   elements.forEach((element) => {
      element.addEventListener('input', saveInfo);
   })
}
```
And then you'd call it like this:
```
saveAllInfos([username, email, text], saveInfo)
```
This would make your code cleaner 🧼 and easier to read 📖